core/org.openjdk.jmc.common/src/main/java/org/openjdk/jmc/common/item/Aggregators.java

```diff
@@ -1048,6 +1048,31 @@ public SetConsumer<T> newItemConsumer(IType<IItem> itemType) {
 		return filter(distinctAsString(attribute, ", "), ItemFilters.type(typeId)); //$NON-NLS-1$
 	}
 
+	public static <T> IAggregator<Long, ?> getIdentifier(String typeId, IAttribute<T> attribute) {
+		IAggregator<Set<T>, ?> aggregator = Aggregators.distinct(attribute);
+		aggregator = filter(aggregator, ItemFilters.type(typeId));
+		return Aggregators.valueBuilderAggregator(aggregator, new IValueBuilder<Long, Set<T>>() {
+			@Override
+			public IType<? super Long> getValueType() {
+				return UnitLookup.RAW_LONG;
+			}
+
+			@Override
+			public Long getValue(Set<T> source) {
+				Long value = 0L;
+				if (source.isEmpty()) {
+					return null;
+				} else {
+					Iterator<?> itr = source.iterator();
+					while (itr.hasNext()) {
+						value = (Long) itr.next();
```

Comment:
**question:** Should there be at least a log if the value is seen more than once ?